src/index.ts

```diff
@@ -0,0 +1,135 @@
+import {ClientFunction, Selector} from 'testcafe'
+import {queries} from '@testing-library/dom'
+import type {
+  Options,
+  TestcafeBoundFunction,
+  TestcafeBoundFunctions,
+} from './types'
+
+declare global {
+  interface Window {
+    TestingLibraryDom: typeof queries
+  }
+}
+
+const SELECTOR_TYPE = Selector('')().constructor.name
+
+const withinSelectors = Object.keys(queries).reduce((acc, withinQueryName) => {
+  return {
+    ...acc,
+    [withinQueryName]: new Function(`
+    const els = arguments[0];
+    if(els.length > 1) {
+      throw new Error("within() only works with a single element, found " + els.length);
+    }
+    const el = els[0];
+    const args = Array.from(arguments).slice(1);
+    return window.TestingLibraryDom.within(el).${withinQueryName}.apply(null, args);
+  `),
+  }
+}, {} as Record<keyof typeof queries, (node: Element, ...methodParams: any[]) => any>)
+
+export async function configureOnce(options: Partial<Options>) {
+  const {content} = configure(options)
+  // @ts-ignore
+  await ClientFunction(new Function(content))()
+}
+
+export function configure(options: Partial<Options>) {
+  const configFunction = `
+  window.TestingLibraryDom.configure(${JSON.stringify(options)});
+`
+  return {content: configFunction}
+}
+
+export function within<T>(
+  sel: string | Selector | SelectorPromise | TestcafeBoundFunction<T>,
+): TestcafeBoundFunctions<typeof queries> {
+  if (sel instanceof Function) {
+    return within(sel())
+  }
+  if (isSelector(sel)) {
+    // @ts-ignore
+    return sel.addCustomMethods(withinSelectors, {returnDOMNodes: true})
+  } else if (typeof sel === 'string') {
+    return within(Selector(sel))
+  } else {
+    throw new Error(
+      `"within" only accepts a query (getBy, queryBy, etc), string or testcafe Selector`,
+    )
+  }
+}
+
+function isSelector(sel: any): sel is Selector {
+  return sel.constructor.name === SELECTOR_TYPE
+}
+
+const bindFunction = <T extends keyof typeof queries>(queryName: T) => {
+  const query = queryName.replace('find', 'query') as T
+  return Selector(
+    (...args) => {
+      const [matcher, ...rest] = args
```

Comment:
thank you for adding that.  was just thinking about doing that myself.  curious are there tests that cover this though (the ...rest)